pkg/auth/providers/azure/clients/ms_graph_client.go

```diff
@@ -151,14 +153,19 @@ func msGraphTokenHasPermissions(token string) bool {
 	if !ok {
 		return false
 	}
-	roleNames := make([]string, len(roles))
+	permissionPresent := make(map[string]bool)
+	for _, p := range msGraphRequiredPermissions {
+		permissionPresent[p] = false
+	}
 	for _, r := range roles {
 		if name, ok := r.(string); ok {
-			roleNames = append(roleNames, name)
+			if slice.ContainsString(msGraphRequiredPermissions, name) {
```

Comment:
Nit:

Rather than checking if the permission is one of our requiredPermissions, you could just throw it in the map. This would let us skip the "slice search and string comparison" part of this step. As with the above nit, this is relatively minor.